dds/idl/marshal_generator.cpp

```diff
@@ -648,39 +810,102 @@ namespace {
-            << streamAndCheck(">> fa", 4) <<
-            "    " << typedefname << "_copy(seq[i], tmp.in());\n";
+            "      " << typedefname << "_forany fa = tmp.inout();\n"
+            << "    if (!(strm >> fa)) {\n";
         } else if (elem_cls & CL_STRING) {
           if (elem_cls & CL_BOUNDED) {
-            const string args = string("seq[i]") + (use_cxx11 ? ", " : ".out(), ")
-              + bounded_arg(elem);
-            be_global->impl_ <<
-              streamAndCheck(">> " + getWrapper(args, elem, WD_INPUT), 4);
-          } else { // unbounded string
+            const string args = string("seq[i]") + (use_cxx11 ? ", " : ".out(), ") + bounded_arg(elem);
+            be_global->impl_ << "    if (!(strm " << ">> " << getWrapper(args, elem, WD_INPUT) << ")) {\n";
+          }
+          else {
             const string getbuffer =
               (be_global->language_mapping() == BE_GlobalData::LANGMAP_NONE)
               ? ".get_buffer()" : "";
-            be_global->impl_ << streamAndCheck(">> seq" + getbuffer + "[i]", 4);
+            be_global->impl_ << "if (!(strm >> seq" + getbuffer + "[i])) {\n";
           }
         } else if (use_cxx11 && (elem_cls & (CL_ARRAY | CL_SEQUENCE))) {
           be_global->impl_ <<
-            streamAndCheck(">> IDL::DistinctType<" + cxx_elem + ", " +
-                           elem_underscores + "_tag>(seq[i])", 4);
-        } else { // Enum, Struct, Union, non-C++11 Array, non-C++11 Sequence
-          be_global->impl_ << streamAndCheck(">> seq[i]", 4);
+          "    if (!(strm >> IDL::DistinctType<" << cxx_elem << ", "  <<
+                           elem_underscores << "_tag>(seq[i])" << ")) {\n";
+        } else {
+          be_global->impl_ << "   if (!(strm >> seq[i])) {\n";
         }
-        be_global->impl_ <<
-          "  }\n"
-          "  return true;\n";
+        std::string seq_resize_func = (use_cxx11) ? "resize" : "length";
+
+        if (try_construct == tryconstructfailaction_use_default) {
+          be_global->impl_ << "     " << type_to_default(elem, "seq[i]") <<
+                              "           strm.set_construction_status(Serializer::ConstructionSuccessful);\n";
+        } else if ((try_construct == tryconstructfailaction_trim) && (elem_cls & CL_BOUNDED) &&
+                   ((elem_cls & CL_STRING) || (elem_cls & CL_SEQUENCE))) {
+          if (elem_cls & CL_STRING){
+            string check_not_empty = (use_cxx11) ? "!seq[i].empty()" : "seq[i].in()";
+            string get_length = (use_cxx11) ? "seq[i].length()" : "ACE_OS::strlen(seq[i].in())";
+            string inout = (use_cxx11) ? "" : ".inout()";
+            if (use_cxx11 ){
+            be_global->impl_ <<
+              "      if (strm.good_bit() && " << check_not_empty << " && ("
+                       << bounded_arg(elem) << " < " << get_length << ")) {\n"
+              "        seq[i]" << inout <<inout << ".resize(" << bounded_arg(elem) <<  ");\n"
+              "      }";
```

Comment:
```suggestion
              be_global->impl_ <<
                "      if (strm.good_bit() && " << check_not_empty << " && ("
                         << bounded_arg(elem) << " < " << get_length << ")) {\n"
                "        seq[i]" << inout <<inout << ".resize(" << bounded_arg(elem) <<  ");\n"
                "      }";
```

Same with other part of the if block.